WS play.ws.ssl.loose.disableHostnameVerification=true doesn't work?

I have set `play.ws.ssl.loose.disableHostnameVerification=true` and then call a web service on localhost that is known to return a certificate for not-localhost (See [this async-http-client issue](https://github.com/AsyncHttpClient/async-http-client/issues/927) for the details), but the call still fails as per the log extract below. 

Windows 8, Play 2.4.1, Scala 2.11.7, Java 8.0_45. 

```
C:\work\play_web>activator run
[info] Loading project definition from C:\work\play_web\project
[info] Set current project to play_web (in build file:/C:/work/play_web/)
Warning: node.js detection failed, sbt will use the Rhino based Trireme JavaScript engine instead to run JavaScript assets compilation, which in some
cases may be orders of magnitude slower than using node.js.

--- (Running the application, auto-reloading is enabled) ---

[info] p.a.l.c.ActorSystemProvider - Starting application default Akka system: application
[info] p.c.s.NettyServer - Listening for HTTP on /0:0:0:0:0:0:0:0:9000

(Server started, use Ctrl+D to stop and go back to the console...)

[info] Compiling 1 Scala source to C:\work\play_web\target\scala-2.11\classes...
WARN application - Logger configuration in conf files is deprecated and has no effect. Use a logback configuration file instead.
INFO application - Creating Pool for datasource 'default'
INFO p.a.d.DefaultDBApi - Database [default] connected at jdbc:mysql://localhost/play_dev?useUnicode=yes&characterEncoding=UTF-8&connectionCollation=utf8_general_ci
INFO p.a.l.c.ActorSystemProvider - Starting application default Akka system: application
INFO play.api.Play - Application started (Dev)
WARN p.a.l.w.n.NingAsyncHttpClientConfigBuilder - buildHostnameVerifier: disabling hostname verification
ERROR application - elb failed to fetch example.com https://localhost/monit
java.net.ConnectException: General SSLEngine problem
        at com.ning.http.client.providers.netty.request.NettyConnectListener.onFutureFailure(NettyConnectListener.java:128) ~[async-http-client-1.9.21.jar:na]
        at com.ning.http.client.providers.netty.request.NettyConnectListener.access$200(NettyConnectListener.java:37) ~[async-http-client-1.9.21.jar:na]
        at com.ning.http.client.providers.netty.request.NettyConnectListener$1.operationComplete(NettyConnectListener.java:101) ~[async-http-client-1.9.21.jar:na]
        at org.jboss.netty.channel.DefaultChannelFuture.notifyListener(DefaultChannelFuture.java:409) ~[netty-3.10.3.Final.jar:na]
        at org.jboss.netty.channel.DefaultChannelFuture.notifyListeners(DefaultChannelFuture.java:395) ~[netty-3.10.3.Final.jar:na]
Caused by: javax.net.ssl.SSLHandshakeException: General SSLEngine problem
        at sun.security.ssl.Handshaker.checkThrown(Handshaker.java:1431) ~[na:1.8.0_45]
        at sun.security.ssl.SSLEngineImpl.checkTaskThrown(SSLEngineImpl.java:535) ~[na:1.8.0_45]
        at sun.security.ssl.SSLEngineImpl.readNetRecord(SSLEngineImpl.java:813) ~[na:1.8.0_45]
        at sun.security.ssl.SSLEngineImpl.unwrap(SSLEngineImpl.java:781) ~[na:1.8.0_45]
        at javax.net.ssl.SSLEngine.unwrap(SSLEngine.java:624) ~[na:1.8.0_45]
Caused by: javax.net.ssl.SSLHandshakeException: General SSLEngine problem
        at sun.security.ssl.Alerts.getSSLException(Alerts.java:192) ~[na:1.8.0_45]
        at sun.security.ssl.SSLEngineImpl.fatal(SSLEngineImpl.java:1728) ~[na:1.8.0_45]
        at sun.security.ssl.Handshaker.fatalSE(Handshaker.java:304) ~[na:1.8.0_45]
        at sun.security.ssl.Handshaker.fatalSE(Handshaker.java:296) ~[na:1.8.0_45]
        at sun.security.ssl.ClientHandshaker.serverCertificate(ClientHandshaker.java:1478) ~[na:1.8.0_45]
Caused by: java.security.cert.CertificateException: No subject alternative DNS name matching localhost found.
        at sun.security.util.HostnameChecker.matchDNS(HostnameChecker.java:204) ~[na:1.8.0_45]
        at sun.security.util.HostnameChecker.match(HostnameChecker.java:95) ~[na:1.8.0_45]
        at sun.security.ssl.X509TrustManagerImpl.checkIdentity(X509TrustManagerImpl.java:455) ~[na:1.8.0_45]
        at sun.security.ssl.X509TrustManagerImpl.checkIdentity(X509TrustManagerImpl.java:436) ~[na:1.8.0_45]
        at sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:252) ~[na:1.8.0_45]		
```

Similarly if I use `play.ws.ssl.hostnameVerifierClass=myutil.play.DisabledHostnameVerifier` - the class is instantiated, but never called. 

```scala
class DisabledHostnameVerifier extends HostnameVerifier {

  Logger.info("Instantiating DisabledHostnameVerifier")

  override def verify(hostname: String, sslSession: SSLSession): Boolean = {
    Logger.info(s"Verifying $hostname as true!")
    true
  }
}
```

```
...
INFO play.api.Play - Application started (Dev)
INFO application - Instantiating DisabledHostnameVerifier
ERROR application - elb failed to fetch example.com https://localhost/monit
java.net.ConnectException: General SSLEngine problem
        at com.ning.http.client.providers.netty.request.NettyConnectListener.onFutureFailure(NettyConnectListener.java:128) ~[async-http-client-1.9.21.jar:na]
        at com.ning.http.client.providers.netty.request.NettyConnectListener.access$200(NettyConnectListener.java:37) ~[async-http-client-1.9.21.jar:na]
        at com.ning.http.client.providers.netty.request.NettyConnectListener$1.operationComplete(NettyConnectListener.java:101) ~[async-http-client-1.9.21.jar:na]
        at org.jboss.netty.channel.DefaultChannelFuture.notifyListener(DefaultChannelFuture.java:409) ~[netty-3.10.3.Final.jar:na]
        at org.jboss.netty.channel.DefaultChannelFuture.notifyListeners(DefaultChannelFuture.java:395) ~[netty-3.10.3.Final.jar:na]
Caused by: javax.net.ssl.SSLHandshakeException: General SSLEngine problem
        at sun.security.ssl.Handshaker.checkThrown(Handshaker.java:1431) ~[na:1.8.0_45]
        at sun.security.ssl.SSLEngineImpl.checkTaskThrown(SSLEngineImpl.java:535) ~[na:1.8.0_45]
        at sun.security.ssl.SSLEngineImpl.readNetRecord(SSLEngineImpl.java:813) ~[na:1.8.0_45]
        at sun.security.ssl.SSLEngineImpl.unwrap(SSLEngineImpl.java:781) ~[na:1.8.0_45]
        at javax.net.ssl.SSLEngine.unwrap(SSLEngine.java:624) ~[na:1.8.0_45]
Caused by: javax.net.ssl.SSLHandshakeException: General SSLEngine problem
        at sun.security.ssl.Alerts.getSSLException(Alerts.java:192) ~[na:1.8.0_45]
        at sun.security.ssl.SSLEngineImpl.fatal(SSLEngineImpl.java:1728) ~[na:1.8.0_45]
        at sun.security.ssl.Handshaker.fatalSE(Handshaker.java:304) ~[na:1.8.0_45]
        at sun.security.ssl.Handshaker.fatalSE(Handshaker.java:296) ~[na:1.8.0_45]
        at sun.security.ssl.ClientHandshaker.serverCertificate(ClientHandshaker.java:1478) ~[na:1.8.0_45]
Caused by: java.security.cert.CertificateException: No subject alternative DNS name matching localhost found.
        at sun.security.util.HostnameChecker.matchDNS(HostnameChecker.java:204) ~[na:1.8.0_45]
        at sun.security.util.HostnameChecker.match(HostnameChecker.java:95) ~[na:1.8.0_45]
        at sun.security.ssl.X509TrustManagerImpl.checkIdentity(X509TrustManagerImpl.java:455) ~[na:1.8.0_45]
        at sun.security.ssl.X509TrustManagerImpl.checkIdentity(X509TrustManagerImpl.java:436) ~[na:1.8.0_45]
        at sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:252) ~[na:1.8.0_45]
```

I experience the same problem in 2.4.6 and 2.5.0-RC1 for `play.ws.ssl.loose.acceptAnyCertificate` config.
I got the following error:
> General SSLEngine problem
Caused by: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target

I tried this workaround in 2.4.6 without success: 
```scala
val sslLooseConfig = new SSLLooseConfig(acceptAnyCertificate = true)
val sslConfig = new SSLConfig(loose = sslLooseConfig)
val config = new WSClientConfig(ssl = sslConfig)
val builder = new NingAsyncHttpClientConfigBuilder(config)
val ws = new play.api.libs.ws.ning.NingWSClient(builder.build())
```

--- a/framework/src/play-ws/src/main/scala/play/api/libs/ws/ahc/AhcConfig.scala
+++ b/framework/src/play-ws/src/main/scala/play/api/libs/ws/ahc/AhcConfig.scala
@@ -7,17 +7,17 @@ package play.api.libs.ws.ahc
 
 import java.security.KeyStore
 import java.security.cert.CertPathValidatorException
-import javax.inject.{ Singleton, Inject, Provider }
-
-import org.asynchttpclient.netty.ssl.JsseSslEngineFactory
-import org.slf4j.LoggerFactory
-
-import org.asynchttpclient.{ DefaultAsyncHttpClientConfig, AsyncHttpClientConfig }
-
+import javax.inject.{ Inject, Provider, Singleton }
 import javax.net.ssl._
-import play.api.{ ConfigLoader, Environment, Configuration }
-import play.api.libs.ws.ssl._
+
+import io.netty.handler.ssl.SslContextBuilder
+import io.netty.handler.ssl.util.InsecureTrustManagerFactory
+import org.asynchttpclient.netty.ssl.JsseSslEngineFactory
+import org.asynchttpclient.{ AsyncHttpClientConfig, DefaultAsyncHttpClientConfig }
+import org.slf4j.LoggerFactory
 import play.api.libs.ws.WSClientConfig
+import play.api.libs.ws.ssl._
+import play.api.{ ConfigLoader, Configuration, Environment }
 
 import scala.concurrent.duration._
 
@@ -279,7 +279,13 @@ class AhcConfigBuilder(ahcConfig: AhcWSClientConfig = AhcWSClientConfig()) {
 
     builder.setAcceptAnyCertificate(sslConfig.loose.acceptAnyCertificate)
 
-    builder.setSslEngineFactory(new JsseSslEngineFactory(sslContext))
+    // If you wan't to accept any certificate you also want to use a loose netty based loose SslContext
+    // Never use this in production.
+    if (sslConfig.loose.acceptAnyCertificate) {
+      builder.setSslContext(SslContextBuilder.forClient().trustManager(InsecureTrustManagerFactory.INSTANCE).build())
+    } else {
+      builder.setSslEngineFactory(new JsseSslEngineFactory(sslContext))
+    }
   }
 
   def buildKeyManagerFactory(ssl: SSLConfig): KeyManagerFactoryWrapper = {

